### ISSUE TITLE
Expose script services to scripts

### ISSUE THREAD

**Issue by adamralph** [scriptcs/scriptcs#960]
`IConsole`, `ILogger`, etc.

E.g. exposing `IConsole ` would allow messages from scripts to be captured in the same way as messages from scriptcs.exe. This means they will be written to the output file when `-output` is specified and also, if a module introduces a custom console, scripts can also write to it.

**Comment by filipw** [comment]
there's no `IScriptServices` :smile:

**Comment by filipw** [comment]
also i think this should be on `ScriptHost` not on the interface; if I have a custom host I probably don't wanna care about setting up and wiring all the services and interfaces

**Comment by filipw** [comment]
one more thought; attempting to expose `ScriptServices` on the host will surely result in circular dependency - since script services contains stuff like REPL which contains script host factory

**Comment by glennblock** [comment]
Agreed there is no IScriptServices, but we should add it. Agreed on
ScriptHost.

**Comment by adamralph** [comment]
@filipw the circular dep is a good point. Moreover, I think `IScriptServices` should be a dedicated interface for this purpose and crafted appropriately, only exposing the things that we think scripts need. It should not just be 'everything that `ScriptServices currently exposes`'. To that end, maybe we should name it something other than `IScriptServices`, to avoid confusion. Or we could rename `ScriptServices`.

In any case, these are implementation details, the first port of call should be the proposed scripting API, so what is the name of the member we want to expose on `ScriptHost`?

Ideas:

```C#
// myscript.csx
Services.Console.WriteLine("hi");       // 1
ScriptServices.Console.WriteLine("hi"); // 2
Env.Console.WriteLine("hi");            // 3
My.Console.WriteLine("hi");             // 4
Script.Console.WriteLine("hi");         // 5
```

In the scripting world, brevity is important so I'm inclined to discount 1 and 2. I like 3 since it reuses the familiar `Env` member. 4 is the shortest, but smells of VB :wink:. 5 is nice, and perhaps my favourite. If we went with that, I would even suggest deprecating `Env` and moving the args to `Script.Args`.

**Comment by filipw** [comment]
yeah I agree.

    Script.Console
    Script.Logger 
    Script.FileSystem

sound good to me

**Comment by glennblock** [comment]
Looks like a good start
